pkgs/development/tools/nasmfmt/default.nix

```diff
@@ -1,22 +1,27 @@
-{ lib, buildGoPackage, fetchFromGitHub, go }:
+{ lib, buildGoModule, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "nasmfmt";
-  version = "unstable-2021-04-24";
+  version = "unstable-2022-09-15";
 
   src = fetchFromGitHub {
     owner = "yamnikov-oleg";
     repo = "nasmfmt";
-    rev = "efba220c5252eb717f080d266dcc8304efdeab40";
-    sha256 = "sha256-snhXF+IP0qzl43rKQ0Ugfo1zv3RyNfjxnMpjZEBgPQg=";
+    rev = "127dbe8e72376c67d7dff89010ccfb49fc7b533e";
+    hash = "sha256-1c7ZOdoM0/Us7cnTT3sds2P5pcCedrCfl0GqQBnf9Rk=";
   };
 
-  goPackagePath = "github.com/yamnikov-oleg/nasmfmt";
+  vendorHash = "sha256-pQpattmS9VmO3ZIQUFn66az8GSmB4IvYhTTCFn6SUmo=";
+
+  preBuild = ''
```

Comment:
`postPatch`?

Reply:
Unfortunately, not working.

Reply:
It builds fine here with `s/preBuild/postPatch/`, x86_64-linux NixOS

Reply:
If you set vendorHash to empty, you will see:
```
go: go.mod file not found in current directory or any parent directory; see 'go help modules'
```
Per `buildGoModule`, `package` depends on `go-module`. So `go-module` goes first. But there is no `postPatch` hook in `go-module`. Consequently, there is no `go.mod` file when building `go-module`.